cmdLine/src/main/scala/topgun/cmdline/FileParser.scala

```diff
@@ -12,30 +15,91 @@ class FileParser(file: File, cmdLine: JfrParseCommandLine, totals: Totals, confi
 
   import configuration.{includeStack, includeThread}
 
-
+  var classLoaderKey = 0
 
   def parse(): Unit = {
 
     println(s"*** File $file")
-    var allocationInNewTlab: Boolean = false
-    var allocationOutsideTlab: Boolean = false
-    var methodProfilingSample: Boolean = false
+    var allocationInNewTlab: Boolean = true
+    var allocationOutsideTlab: Boolean = true
+    var methodProfilingSample: Boolean = true
+    var operatingSytem:String = "unknown"
+    var classPaths =  new util.ArrayList[URL]
+
+    def preProcessing(): Unit = {
+
+      def extractClassPath(event: RecordedEvent): Unit = {
+        if(event.getValue("key").toString.contains("java.class.path")) {
+          val clazzpaths:String = event.getValue("value")
+          val delim = if (operatingSytem.equals("win")) ";" else ":"
+          clazzpaths.split(delim).foreach(path => {classPaths.add(Paths.get(path).toUri.toURL)})
+        }
+      }
+
+      def extractJfrSettings(event: RecordedEvent): Unit = {
+        val id:Long = event.getValue("id")
+        val name:String = event.getValue("name")
+        val value:String = event.getValue("value")
+        if( name.equals("enabled") ) {
+          id match {
+            //Allocation in new TLAB // id = 331
+            case 331 => allocationInNewTlab = value.toBoolean;
+            //Allocation outside TLAB // id = 332
+            case 332 => allocationOutsideTlab = value.toBoolean
+            //Method Profiling Sample // id = 353
+            case 353 => methodProfilingSample = value.toBoolean
+            case _ =>
+          }
+        }
+      }
+
+      def extractOperatingSystemInfo(event: RecordedEvent): Unit={
+        val osInfo = event.getValue("osVersion").toString.toLowerCase()
+        if(osInfo.contains(osInfo)){
+          operatingSytem = "win"
+        }else{
+          operatingSytem = "others"
+        }
+      }
+
+      val recordingFile = new RecordingFile(file.toPath)
+      while(recordingFile.hasMoreEvents){
+        val event = recordingFile.readEvent()
+        event.getEventType.getLabel match {
+          case "Initial System Property" => extractClassPath(event)
+          case "Recording Setting" => extractJfrSettings(event);
+          case "OS Information" => extractOperatingSystemInfo(event);
+          case _ =>
+        }
+      }
+      recordingFile.close()
+
+      def validateJfr():Unit={
+        try{
```

Comment:
you dont need a try/catch here
1.  its the default behaviour
2. you make it impossible to test

Reply:
done